tests/python/test_no_activate.py

```diff
@@ -0,0 +1,30 @@
+import taichi as ti
+
+
+@ti.require(ti.extension.sparse)
+@ti.all_archs
+def test_no_activate():
+    x = ti.var(ti.f32)
+
+    n = 1024
+
+    d = ti.root.dynamic(ti.i, n, chunk_size=32)
+    d.place(x)
+
+    @ti.kernel
+    def initialize():
+        for i in range(n):
+            x[i] = 1
+
+    @ti.kernel
+    def func():
+        ti.no_activate(d)
+        for i in range(n // 2):
```

Comment:
> if we use `for i in x` here and then **write** to `x[i]`, is it a strong or weak access? That is, is Taichi's runtime smart enough to know that we've done a series of listgens, so that `i` will always be activated?

That would be a *weak* access. We have a pass to weaken them: https://github.com/taichi-dev/taichi/blob/c7496d8ac96e9ff4b9ccf9684f78ee8b95e7e0aa/taichi/transforms/flag_access.cpp#L64